sormas-backend/src/main/java/de/symeda/sormas/backend/labmessage/LabMessageFacadeEjb.java

```diff
@@ -0,0 +1,94 @@
+		target.setPersonBirthDateDD(source.getPersonBirthDateDD());
+		target.setPersonBirthDateMM(source.getPersonBirthDateMM());
+		target.setPersonBirthDateYYYY(source.getPersonBirthDateYYYY());
+		target.setPersonCity(source.getPersonCity());
+		target.setPersonFirstName(source.getPersonFirstName());
+		target.setPersonHouseNumber(source.getPersonHouseNumber());
+		target.setPersonLastName(source.getPersonLastName());
+		target.setPersonPostalCode(source.getPersonPostalCode());
+		target.setPersonSex(source.getPersonSex());
+		target.setPersonStreet(source.getPersonStreet());
+		target.setProcessed(source.isProcessed());
+		target.setSampleDateTime(source.getSampleDateTime());
+		target.setSampleMaterial(source.getSampleMaterial());
+		target.setSampleReceivedDate(source.getSampleReceivedDate());
+		target.setSpecimenCondition(source.getSpecimenCondition());
+		target.setTestDateTime(source.getTestDateTime());
+		target.setTestedDisease(source.getTestedDisease());
+		target.setTestLabCity(source.getTestLabCity());
+		target.setTestLabExternalId(source.getTestLabExternalId());
+		target.setTestLabName(source.getTestLabName());
+		target.setTestLabPostalCode(source.getTestLabPostalCode());
+		target.setTestResult(source.getTestResult());
+		target.setTestType(source.getTestType());
+
+		return target;
+	}
+
+	public LabMessageDto toDto(LabMessage source) {
+
+		if (source == null) {
+			return null;
+		}
+		LabMessageDto target = new LabMessageDto();
+		DtoHelper.fillDto(target, source);
+
+		target.setLabMessageDetails(source.getLabMessageDetails());
+		target.setLabSampleId(source.getLabSampleId());
+		target.setPersonBirthDateDD(source.getPersonBirthDateDD());
+		target.setPersonBirthDateMM(source.getPersonBirthDateMM());
+		target.setPersonBirthDateYYYY(source.getPersonBirthDateYYYY());
+		target.setPersonCity(source.getPersonCity());
+		target.setPersonFirstName(source.getPersonFirstName());
+		target.setPersonHouseNumber(source.getPersonHouseNumber());
+		target.setPersonLastName(source.getPersonLastName());
+		target.setPersonPostalCode(source.getPersonPostalCode());
+		target.setPersonSex(source.getPersonSex());
+		target.setPersonStreet(source.getPersonStreet());
+		target.setProcessed(source.isProcessed());
+		target.setSampleDateTime(source.getSampleDateTime());
+		target.setSampleMaterial(source.getSampleMaterial());
+		target.setSampleReceivedDate(source.getSampleReceivedDate());
+		target.setSpecimenCondition(source.getSpecimenCondition());
+		target.setTestDateTime(source.getTestDateTime());
+		target.setTestedDisease(source.getTestedDisease());
+		target.setTestLabCity(source.getTestLabCity());
+		target.setTestLabExternalId(source.getTestLabExternalId());
+		target.setTestLabName(source.getTestLabName());
+		target.setTestLabPostalCode(source.getTestLabPostalCode());
+		target.setTestResult(source.getTestResult());
+		target.setTestType(source.getTestType());
+
+		return target;
+	}
+
```

Comment:
Is this complete now? I'm wondering because the other facades I checked (and so also the facadeEjbs) have a saveContact/savePerson/save... which then calls a ensurePersisted(...) method. Is this standard behaviour which should also be implemented for the labMessages?

What about a getByUuid()?

Reply:
A standard set of methods doesn't exist. Normally you would add methods only if they are needed. Nevertheless, I will add the save method, since it is needed in the ExternalLabFacade.